packages/mui-material/src/Button/Button.js

```diff
@@ -504,6 +718,10 @@ Button.propTypes /* remove-proptypes */ = {
    * Element placed before the children.
    */
   startIcon: PropTypes.node,
+  /**
+   * Styles applied to the startIcon element if `loading={true}` and `loadingPosition="start"`.
+   */
+  startIconLoadingStart: PropTypes.string,
```

Comment:
This prop shouldn't be here. These is part of the `classes` prop which is already handled in [buttonClasses.ts](https://github.com/mui/material-ui/pull/42987/files#diff-678f4576d82534ecac84d6c93e8166bad24d5d5aa804c61f4896853ca433631a)